src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -212,52 +202,40 @@ internal static int Encrypt(SafeSslHandle context, byte[] input, int offset, int
                     case Ssl.SslErrorCode.SSL_ERROR_ZERO_RETURN:
                     case Ssl.SslErrorCode.SSL_ERROR_WANT_READ:
                         break;
-
+                    //indicates we need to write the out buffer and write again
+                    case Ssl.SslErrorCode.SSL_ERROR_WANT_WRITE:
+                        break;
                     default:
                         throw new SslException(SR.Format(SR.net_ssl_encrypt_failed, errorCode), innerError);
                 }
             }
-            else
-            {
-                int capacityNeeded = Crypto.BioCtrlPending(context.OutputBio);
-
-                if (output == null || output.Length < capacityNeeded)
-                {
-                    output = new byte[capacityNeeded];
-                }
-
-                retVal = BioRead(context.OutputBio, output, capacityNeeded);
-            }
-
-            return retVal;
+            return context.OutputBio.TakeBytes(out output);
```

Comment:
Yeah so it doesn't need to be a ref anymore, it always returns the same buffer (eg the out is always the same). The out is because of the DoHandshake using the same method and that will always get a new buffer. Maybe it should be a ref as well. I didn't want to make a GetBytes(out buffer) and GetBytes() but I guess that an overload like that would make more sense. On the method signature for the PAL it could actually have the ref/out removed, except that it is common with the other two PAL's